frontends/p4/tableKeyNames.cpp

```diff
@@ -32,10 +38,16 @@ class KeyNameGenerator : public Inspector {
 
     void postorder(const IR::Expression* expression) override { error(expression); }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("foo.bar")` for
```

Comment:
For this pass, I'd like to see examples of the names generated from the IR.  This is much faster than, say, digging through the parser to figure out how syntax corresponds to IR objects.

For example, `_` is apparently a valid PathExpression, but---looking at the parser---will not appear in an expression inside a KeyElement.  (In fact, it's not obvious to me why the `Path` class has an `isDontCare()` method.)  On the other hand, dot-prefixed identifiers are also valid PathExpressions and may indeed appear here, in which case the name generated includes the "." prefix.

This is not obvious from the implementation of the postorder method.

I don't have a strong preference as to whether this lives as part of comments on pre-/postorder methods, on the `name` field of the `KeyNameGenerator`, or as part of the class comment for the pass itself.